### PR TITLE
feat(spanner): use multiple channels instead of multiple client for the multiple rows cpu benchmark

### LINKED ISSUE #4032: update benchmarks to use multiple Channels instead of multiple Clients
Since the `SessionPool` now supports multiple Channels for a single `Client`, change the benchmarks to use one `Client` with multiple channels instead of multiple clients.

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #4032

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6659)

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]
Seems to me there are two changes in this PR:

- Some refactoring to make the benchmark "cleaner"
- The `clients` to `channels` rename / changes.

It would be easier to grok the changes if you broke them down into smaller PRs.  Is it possible to do so?

**Comment by dbolduc** [comment]
You are correct. I removed the refactoring commit. I will refactor in a subsequent PR.

File: google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc [hunk 1]
```diff
@@ -1035,6 +1018,7 @@ class SelectExperiment : public Experiment {
  *
  * The values of K, M, N are configurable.
  */
+
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc:1021]: nit: we normally do not write these.

File: google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc [hunk 2]
```diff
@@ -441,10 +437,40 @@ class ExperimentImpl {
...
+              << std::flush;
+
+    auto connection = spanner::MakeConnection(
+        database, spanner::ConnectionOptions().set_num_channels(num_channels),
+        // This pre-creates all the Sessions we will need (one per thread).
+        spanner::SessionPoolOptions().set_min_sessions(config.maximum_threads));
+    return spanner::Client(std::move(connection));
+  }
+
+  std::vector<std::shared_ptr<spanner_internal::SpannerStub>> MakeStubs(
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc:460]: Splitting the function is much cleaner, thanks.

File: google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc [hunk 3]
```diff
@@ -797,6 +789,7 @@ class ReadExperiment : public Experiment {
  *     `config.query_size` rows
  *   - Measure the CPU time required by the previous step
  */
+
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc:792]: nit: we normally do not put blank lines here (there are other new blank lines too?).

File: google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc [hunk 4]
```diff
@@ -1315,36 +1290,24 @@ class MutationExperiment : public Experiment {
 
   Status SetUp(Config const&, spanner::Database const&) override { return {}; }
 
-  Status TearDown(Config const&, spanner::Database const&) override {
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/multiple_rows_cpu_benchmark.cc:1318]: This change seems good, but unrelated?

**Review by coryan — APPROVED** [review]
A few nits that I trust you can fix before merging.

**Review by dbolduc — COMMENTED** [review]
> Dismissed @coryan from a discussion.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6659)* status: 0 of 6 files reviewed, all discussions resolved (waiting on @dbolduc)

**Pull request merged**